bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -1695,30 +1724,34 @@ bool BCStateTran::onMessage(const FetchResPagesMsg *m, uint32_t msgLen, uint16_t
 
   FetchingState fetchingState = getFetchingState();
 
-  // if msg should be rejected
-  if (fetchingState != FetchingState::NotFetching || !psd_->hasCheckpointDesc(m->requiredCheckpointNum)) {
-    RejectFetchingMsg outMsg;
-    outMsg.requestMsgSeqNum = m->msgSeqNum;
-
-    LOG_WARN(logger_,
-             "Rejecting msg. Sending RejectFetchingMsg to replica " << KVLOG(replicaId,
-                                                                             fetchingState,
-                                                                             outMsg.requestMsgSeqNum,
-                                                                             m->msgSeqNum,
-                                                                             m->lastCheckpointKnownToRequester,
-                                                                             m->requiredCheckpointNum,
-                                                                             m->lastKnownChunk));
-
-    metrics_.sent_reject_fetch_msg_++;
-
-    replicaForStateTransfer_->sendStateTransferMessage(
-        reinterpret_cast<char *>(&outMsg), sizeof(RejectFetchingMsg), replicaId);
+  auto sendRejectFetchingMsg = [&](const char *reason) {
+    // if msg should be rejected
+    if (fetchingState != FetchingState::NotFetching || !psd_->hasCheckpointDesc(m->requiredCheckpointNum)) {
+      RejectFetchingMsg outMsg;
+      outMsg.requestMsgSeqNum = m->msgSeqNum;
 
+      LOG_WARN(logger_,
+               "Rejecting msg. Sending RejectFetchingMsg to replica " << KVLOG(reason,
+                                                                               replicaId,
+                                                                               fetchingState,
+                                                                               outMsg.requestMsgSeqNum,
+                                                                               m->msgSeqNum,
+                                                                               m->lastCheckpointKnownToRequester,
+                                                                               m->requiredCheckpointNum,
+                                                                               m->lastKnownChunk));
+
+      metrics_.sent_reject_fetch_msg_++;
+
+      replicaForStateTransfer_->sendStateTransferMessage(
+          reinterpret_cast<char *>(&outMsg), sizeof(RejectFetchingMsg), replicaId);
+    }
     return false;
```

Comment:
This has become a lambda now and I think the result should be observed when calling it.